model/flow/address.go

```diff
@@ -13,9 +13,10 @@ import (
 type Address [AddressLength]byte
 
 type AddressGenerator interface {
-	NextAddress() (Address, error)
+	NextAddress() (Address, uint64, error) // returning address, index and error
```

Comment:
What is the reason of choosing to add a third output for the index instead of using a separate function to return the index? 
I can see that the returned index is ignored most of the time, so perhaps calling a dedicated index function when needed would be cleaner?

Reply:
for performance reasons, few method calls. maybe doesn't worth it

Reply:
going to remove this param and revert changes.

Reply:
Basically this prevents few function calls and one read value from the ledger cache.

Reply:
I reverted these changes

Reply:
I see your point. It probably saves time if we need the returned index every time we get the address.